charts/mattermost-push-proxy/templates/deployment.yaml

```diff
@@ -41,104 +41,68 @@ spec:
               name: push-config-template
               subPath: push-config.json
-          {{- if .Values.applePushSettings.authKey }}
+            # Volume Mounts handling based on externalSecrets.enabled
             - mountPath: {{ .Values.applePushSettings.authKeyFile }}
               name: apple-push-auth-key
               subPath: {{ .Values.applePushSettings.authKeyFileName }}
-          {{- end }}
-          {{- if .Values.applePushSettings.apple.privateCert }}
             - mountPath: /certs/apple-push-cert.pem
               name: apple-push-cert
               subPath: apple-push-cert.pem
-          {{- end }}
-          {{- if .Values.applePushSettings.apple_rn.privateCert }}
             - mountPath: /certs/apple-rn-push-cert.pem
               name: apple-rn-push-cert
               subPath: apple-rn-push-cert.pem
-          {{- end }}
-          {{- if .Values.applePushSettings.apple_rnbeta.privateCert }}
             - mountPath: /certs/apple-rnbeta-push-cert.pem
               name: apple-rnbeta-push-cert
               subPath: apple-rnbeta-push-cert.pem
-          {{- end }}
-          {{- if .Values.androidPushSettings.android.serviceFile }}
             - mountPath: "{{ .Values.androidPushSettings.android.serviceFileLocation }}"
               name: android-service-file
               subPath: "{{ .Values.androidPushSettings.android.serviceFileName }}"
-          {{- end }}
-          {{- if .Values.androidPushSettings.android_rn.serviceFile }}
             - mountPath: "{{ .Values.androidPushSettings.android_rn.serviceFileLocation }}"
               name: android-rn-service-file
               subPath: "{{ .Values.androidPushSettings.android_rn.serviceFileName }}"
-          {{- end }}
           resources:
-{{ toYaml .Values.resources | indent 12 }}
-    {{- with .Values.nodeSelector }}
-      nodeSelector:
-        {{- toYaml . | nindent 8 }}
-      {{- end }}
-    {{- with .Values.affinity }}
-      affinity:
-        {{- toYaml . | nindent 8 }}
-    {{- end }}
-    {{- with .Values.tolerations }}
-      tolerations:
-        {{- toYaml . | nindent 8 }}
-    {{- end }}
+{{ toYaml .Values.resources | nindent 10 }}
       volumes:
         - name: push-config-template
           configMap:
             name: {{ include "mattermost-push-proxy.fullname" . }}-push-config-template
             items:
-            - key: push-config.json
-              path: push-config.json
-        {{- if .Values.applePushSettings.authKey }}
+              - key: push-config.json
+                path: push-config.json
+        # Secret definitions based on externalSecrets.enabled
         - name: apple-push-auth-key
           secret:
-            secretName: {{ include "mattermost-push-proxy.fullname" . }}-apple-certs
+            secretName: {{ if .Values.externalSecrets.enabled }}{{ include "mattermost-push-proxy.fullname" . }}-secret{{ else }}{{ include "mattermost-push-proxy.fullname" . }}-apple-certs{{ end }}
```

Comment:
With the  removal of {{- if .Values.applePushSettings.authKey }} the secret will be create even if the values are empty, which we don't want right?

Reply:
@stylianosrigas This change has the purpose when we have external secrets in place all key values to be under one AWS secrets for cost/simplicity reasons, at the same time for backwards compatibility we need to keep the previous structure exactly as is when we have not enable external secrets. That is the purpose of this if clause.

Reply:
I am referring to the removal of  {{- if .Values.applePushSettings.authKey }}. This is now replaced by a single else which means that if external secrets are disabled it will create the old type secrets even if the applePushSettings.authKey is empty. Previously using an empty applePushSettings.authKey  would not generate any secret.

Reply:
thanks for the catch.